### PR TITLE
[Feature] Implement LocalMetastore

### PR DESCRIPTION
**Pull request by stephen-shelby**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

related with https://github.com/StarRocks/starrocks/issues/6211

related with https://github.com/StarRocks/starrocks/issues/5062

## Problem Summary(Required) ：

Extract the metadata related from GlobalMetastore to LocalMetastore.
LocalMetastore will implement ConnectorMetadata API.
The next pr will remove the metadata related to GlobalStateMgr and use LocalMetastore.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 20 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/connector/ConnectorMetadata.java|0|18|00.00%|

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 1]
```diff
@@ -0,0 +1,3973 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:1]: if this file is most from other files, you should keep the original license header.

Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java]: 1

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 20 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/connector/ConnectorMetadata.java|0|18|00.00%|

**Comment by gengjun-git** [comment]
Please add some description of this PR

**Comment by stephen-shelby** [comment]
done

**Comment by stephen-shelby** [comment]
> Please add some description of this PR

done

**Review by gengjun-git — APPROVED** [review]

**Review by caneGuy — APPROVED** [review]

**Pull request merged**